lib/eco/autolboxdetector.py

```diff
@@ -97,16 +97,22 @@ def detect_lbox_h2(self, errornode):
                     left = parent.get_attr("children", pv)[0] # bos
                 else:
                     left = parent.get_attr("left", pv)
-                if left:
+                if left and len(left.get_attr("children", pv)) > 0:
                     state = left.state
                     element = self.op.syntaxtable.lookup(state, lbox)
                     if type(element) in [Reduce, Shift]:
                         term = self.find_terminal(left, pv)
                         if type(term) is EOS:
+                            parent = parent.get_attr("parent", pv)
                             continue
                         while term and term.lookup in ws:
                             # skip whitespace
                             term = term.next_term
+                        element = self.op.syntaxtable.lookup(term.state, lbox)
+                        if type(element) not in [Reduce, Shift]:
+                            # can we still shift lbox?
```

Comment:
What does the comment mean?

Reply:
We have already checked that the lbox can be shifted after the nonterminal `left`. This is a sanity check, that tests if lbox can still be shifted after `term` which is the most-right terminal in `left`. This should almost always be true, so this double checks that this is really the case. I believe there can be cases, where the terminal is an error node, in which case this second check will fail, so that we don't insert a box which then turns out to be invalid when reparsed.

Reply:
Do we have a test case for this? If not, can we maybe expand the comment along the lines of what you've just written.

Reply:
I think I need to add some tests for this and the other features, as I just found another bug with the new stuff.

Reply:
Although in this case, I don't currently have a test, though I might be able to come up with one.

Reply:
Let's give it a go.